app/models/client.rb

```diff
@@ -6,6 +6,15 @@ def full_name
     "#{first_name} #{last_name}"
   end
 
+  def contacted_at
+    # the date of the most recent message sent or received
+    last_message = Message.where(client: self).order(:created_at).last
+    if last_message
+      return last_message.created_at
+    end
+    self.updated_at
+  end
```

Comment:
This introduces a N+1 query when sorting the list of clients on the other page. One thought on mitigating this is to use `touch: true` on the `belongs_to :client` association in `Message`. This will update the Client object's `updated_at` field whenever a new `Message` is created. Slight performance penalty on creation of the message, but possibly worth it to have a better idea of when the client was updated most recently without having to join tables.